### PR TITLE
Docs reorg and added Async connection page

### PR DESCRIPTION
**Pull request by rozza**
JAVA-1728 JAVA-1732

[Review on Reviewable](https://reviewable.io/reviews/rozza/mongo-java-driver/82)

### CODE REVIEW

File: docs/reference/content/driver/getting-started/installation-guide.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+[menu.main]
+  parent = "Sync Getting Started"
+  identifier = "Sync Installation Guide"
+  weight = 1
+  pre = "<i class='fa'></i>"
++++
+
+# Installation
+
+There are two different MongoDB drivers available in the 3.0 release.  The preferred method is to use the new MongoDB Driver as a
```
Comment (jyemin) [inline docs/reference/content/driver/getting-started/installation-guide.md]: Just reading this again, it's confusing.  Especially now that we talk about two drivers: driver and async driver.

How about:

There are two Maven artifacts available in the 3.0 release.  The preferred artifact for new applications is mongodb-driver; however, we still publish the legacy mongo-java-driver uber-jar.

File: docs/reference/content/bson/installation-guide.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+[menu.main]
+  parent = "BSON"
+  weight = 50
+  pre = "<i class='fa'></i>"
++++
+
+# Installation
+
+The BSON library is a required dependency of all the MongoDB Java Drivers and if using a  dependency management system, it should be 
+automatically installed alongside the driver. However, it can be used as a standalone library ad
```
Comment (jyemin) [inline docs/reference/content/bson/installation-guide.md]: Not sure where you were going with this sentence.  It ends rather abruptly :)

File: docs/reference/content/bson/installation-guide.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+title = "Installation Guide"
+[menu.main]
+  parent = "BSON"
+  weight = 50
+  pre = "<i class='fa'></i>"
++++
+
+# Installation
+
+The BSON library is a required dependency of all the MongoDB Java Drivers and if using a  dependency management system, it should be
```
Comment (jyemin) [inline docs/reference/content/bson/installation-guide.md]: lowercase "Drivers"

Comment (jyemin) [inline docs/reference/content/bson/installation-guide.md]: "will", not "should"

**Comment by jyemin** [comment]
just a few comments.  looks fantastic

**Comment by rozza** [comment]
all cleaned up, merged and rebased.

**Pull request closed without merging**